### DISCUSSION TITLE
.NET 4.7.2 with Wix installer fails with : Failed to verify signature of payload: NetFx472Redist

### DISCUSSION THREAD

**Discussion by rekhasasidharan** [wixtoolset/issues#6551] [Questions] [answered]
After  bundling the my application with .NET framework 4.7.2 and installing it I am seeing an error  

Error 0x80070490: Failed to find expected public key in certificate chain.
[0D20:06EC][2021-08-20T21:22:38]e000: Error 0x80070490: Failed to verify expected payload against actual certificate chain.
[0D20:06EC][2021-08-20T21:22:38]e000: Error 0x80070490: Failed to verify signature of payload: NetFx472Redist
[0D20:06EC][2021-08-20T21:22:38]e310: Failed to verify payload: NetFx472Redist at path: C:\ProgramData\Package Cache\.unverified\NetFx472Redist, error: 0x80070490. Deleting file. 

Appreciate if anyone help me out  to find if I am missing something to add in the project.  Where I need to manage <Chain> element referring the PackageGroupRef. 

The snippet of code I have tried out is 

 string productMsi = project.BuildMsi();

         var bootstrapper =
             new Bundle("My_Installer",
                 new PackageGroupRef("NetFx472Redist"), new MsiPackage(productMsi) { DisplayInternalUI = false, Compressed = true });

         bootstrapper.Version = project.Version;
         bootstrapper.OutFileName = "My_Installer.msi";
         bootstrapper.WxsFiles.Add(@"wix\NetFx472.wxs");
         bootstrapper.Include(WixExtension.NetFx);
         bootstrapper.Include(WixExtension.Util);
         bootstrapper.Build();

And the NetFx472.wxs file is managed as 

<?xml version="1.0" encoding="utf-8"?>

<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">

   

   <?define NetFx472MinRelease = 461808 ?>
   <?define NetFx472WebLink = http://go.microsoft.com/fwlink/?LinkId=863262 ?>
   <?define NetFx472RedistLink = http://go.microsoft.com/fwlink/?LinkId=863265 ?>
   <?define NetFx472EulaLink = http://referencesource.microsoft.com/license.html ?>

   <Fragment>
      <PropertyRef Id="WIXNETFX4RELEASEINSTALLED" />
      <Property Id="WIX_IS_NETFRAMEWORK_472_OR_LATER_INSTALLED" Secure="yes" />
      <SetProperty Id="WIX_IS_NETFRAMEWORK_472_OR_LATER_INSTALLED" Value="1" After="AppSearch">
         WIXNETFX4RELEASEINSTALLED >= "#$(var.NetFx472MinRelease)"
      </SetProperty>
   </Fragment>

   <Fragment>
      <util:RegistrySearchRef Id="NETFRAMEWORK45"/>

      <WixVariable Id="WixMbaPrereqPackageId" Value="NetFx472Web" />
      <WixVariable Id="WixMbaPrereqLicenseUrl" Value="$(var.NetFx472EulaLink)" Overridable="yes" />
      <WixVariable Id="NetFx472WebDetectCondition" Value="NETFRAMEWORK45 &gt;= $(var.NetFx472MinRelease)" Overridable="yes" />
      <WixVariable Id="NetFx472WebInstallCondition" Value="" Overridable="yes" />
      <WixVariable Id="NetFx472WebPackageDirectory" Value="redist\" Overridable="yes" />

      <PackageGroup Id="NetFx472Web">
         <ExePackage
             InstallCommand="/q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;"
             RepairCommand="/q /norestart /repair /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;"
             UninstallCommand="/uninstall /q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;"
             PerMachine="yes"
             DetectCondition="!(wix.NetFx472WebDetectCondition)"
             InstallCondition="!(wix.NetFx472WebInstallCondition)"
             Id="NetFx472Web"
             Vital="yes"
             Permanent="yes"
             Protocol="netfx4"
             DownloadUrl="$(var.NetFx472WebLink)"
             LogPathVariable="NetFx472FullLog"
             Compressed="no"
             Name="!(wix.NetFx472WebPackageDirectory)NDP472-KB4054531-Web.exe">
            <RemotePayload
              CertificatePublicKey="C090C1A2CAFA9B967D9C87C7FE02F7C01FBDE4F2"
              CertificateThumbprint="5EAD300DC7E4D637948ECB0ED829A072BD152E17"
              Description="Microsoft .NET Framework 4.7.2 Setup"
              Hash="507ECDADC23A27C2283BA130A2AA51650E6BC05B"
              ProductName="Microsoft .NET Framework 4.7.2"
              Size="1447320"
              Version="4.7.3062.0" />
         </ExePackage>
      </PackageGroup>
   </Fragment>

   <Fragment>
      <util:RegistrySearchRef Id="NETFRAMEWORK45"/>

      <WixVariable Id="WixMbaPrereqPackageId" Value="NetFx472Redist" />
      <WixVariable Id="WixMbaPrereqLicenseUrl" Value="$(var.NetFx472EulaLink)" Overridable="yes" />
      <WixVariable Id="NetFx472RedistDetectCondition" Value="NETFRAMEWORK45 &gt;= $(var.NetFx472MinRelease)" Overridable="yes" />
      <WixVariable Id="NetFx472RedistInstallCondition" Value="" Overridable="yes" />
      <WixVariable Id="NetFx472RedistPackageDirectory" Value="redist\" Overridable="yes" />

      <PackageGroup Id="NetFx472Redist">
         <ExePackage
             InstallCommand="/q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;"
             RepairCommand="/q /norestart /repair /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;"
             UninstallCommand="/uninstall /q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx472FullLog].html&quot;"
             PerMachine="yes"
             DetectCondition="!(wix.NetFx472RedistDetectCondition)"
             InstallCondition="!(wix.NetFx472RedistInstallCondition)"
             Id="NetFx472Redist"
             Vital="yes"
             Permanent="yes"
             Protocol="netfx4"
             DownloadUrl="$(var.NetFx472RedistLink)"
             LogPathVariable="NetFx472FullLog"
             Compressed="no"
             Name="!(wix.NetFx472RedistPackageDirectory)NDP472-KB4054530-x86-x64-AllOS-ENU.exe">
            <RemotePayload
              CertificatePublicKey="C090C1A2CAFA9B967D9C87C7FE02F7C01FBDE4F2"
              CertificateThumbprint="5EAD300DC7E4D637948ECB0ED829A072BD152E17"
              Description="Microsoft .NET Framework 4.7.2 Setup"
              Hash="D3A416DC5FC75758D41B4C0158ACA69270D2A904"
              ProductName="Microsoft .NET Framework 4.7.2"
              Size="71607232"
              Version="4.7.3062.0" />
         </ExePackage>
      </PackageGroup>
   </Fragment>
</Wix>

**Comment by rseanhall** [comment, marked as answer]
Did you see the existing discussions about this at https://github.com/wixtoolset/issues/discussions/6441 and https://github.com/wixtoolset/issues/discussions/6528?

**Reply by rekhasasidharan** [reply]
Yes. It is helpful .
